packages/tools/replay-tool/src/helpers.ts

```diff
@@ -267,10 +267,18 @@ export async function loadContainer(
         options,
         logger,
     });
-    const container: Container = await loader.resolve({ url: resolved.url });
 
-    assert(container.existing,
-        0x1c4 /* "Container does not exist!" */); // ReplayFileDeltaConnection.create() guarantees that
-
-    return container;
+    return loader.resolve(
+        {
+            url: resolved.url,
+            headers: {
+                [LoaderHeader.clientDetails]: {
+                    // #6346
+                    // hardcoded keyword to be replaced by `LegacyCreateOnLoadEnvironmentKey`
```

Comment:
If this is the only place where LegacyCreateOnLoadEnvironmentKey is needed, then I think we could simplify.
Though maybe in next PR, as it's possible we will learn (once you merge it), through someone complaining that we broke their scenario, that there are more users :)

And by simplify I mean move complexity out from loader into either this layer (replay tool) or into the driver it uses (file driver). We could synthetize empty snapshot for these files on the fly, and thus remove completely that path from loader. Also if that's the only place used, we can already allow ODSP & FRS services to deprecate it, as this tool does not uses any of them - it uses driver that works on top of local files only.

Reply:
Also would be nice to use LegacyCreateOnLoadEnvironmentKey  directly here

Reply:
To your last point, I think we'd need a version bump for it as `LegacyCreateOnLoadEnvironmentKey` is in a different package. Is this something that can be made with one PR without combining the code proposal with the version change? I'd rather separate the two.

For the first part, yeah. That sounds like a great idea. When following the code, that part of the loader did seem like it being used only here. Maybe this is an opportunity to simplify it.

Reply:
Opened https://github.com/microsoft/FluidFramework/issues/6626